Sources/Kitura/CodableRouter.swift

```diff
@@ -600,14 +603,31 @@ extension Router {
         }
     }
 
-    internal func parameterIsPresent(in route: String) -> Bool {
-        if route.contains(":") {
-            let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
-            let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
-            Log.error("Erroneous path '\(route)', parameter ':\(parameter)' is not allowed. Codable routes do not allow parameters.")
+    func pathSyntaxIsValid(_ path: String, identifierExpected: Bool) -> Bool {
+        let identifierSupplied = path.contains(":")
+        switch (identifierExpected, identifierSupplied) {
+        case (false, true):
+            Log.error("Path '\(path)' is not allowed: Codable routes do not allow path parameters.")
+            return false
+        case (true, false):
+            Log.warning("Identifier expected for '\(path)'. :id will be automatically appended to the path.")
+            return true
+        case (true, true):
+            /// Precondition: Path is known to contain a : character
+            func pathHasSingleParamIdAsSuffix(_ path: String) -> Bool {
+                let parameterString = path.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
+                let parameter = parameterString.count > 0 ? parameterString[1] : ""
```

Comment:
I'm talking about this specific function, not the broader code.  You do `path.split()` with `maxSplits: 1`. That means theoretically it could do 0 splits. In which case `parameterString.count` would be 1 and we would crash.

Also, I don't understand why we pass `omittingEmptySubsequences: false`. Why would we want empty subsequences to be created?

Reply:
Yeah it would crash if you call this function on a String which does not have the ```:```  and hence  the comment ```Precondition: Path is known to contain a : character ``` has  been retained. But in  this case it will not happen as we would not end up in calling this function if the path does not have  a ```:``` character and the possibility of ending up in performing ```0``` splits is not there . Also, as this  method's scope is  limited to this ```case``` one will not end up in calling it without the ```contains(":")``` check from outside code. Moreover, ```parameterIsPresent``` in the  existing  code is also doing the same.

Reply:
Right, as I noted, it won't crash today because of the circumstances we call the function under. But who knows about the future? We should practice [Defensive Programming](https://en.wikipedia.org/wiki/Defensive_programming) and not store up problems for the future.

Reply:
If  ```omittingEmptySubsequences```  is  set to false an empty subsequence is returned in the result for each consecutive pair of ```:``` . Its useful in cases of handling the case like ```/status/:``` where empty sub-sequence would be returned and will not end up in crash while doing comparison. 
eg :  In case of  ```omittingEmptySubsequences: false```
```
let path = "/status/:"
let parameterString = path.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
print(parameterString.count)  ------------>  would be 2 
let parameter = parameterString.count > 0 ? parameterString[1] : ""
print(parameter) -----------> would be ""
``` 
In case of ```omittingEmptySubsequences: true``` , an empty sub-sequence would not be returned.
```
let path = "/status/:"
let parameterString = path.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
print(parameterString.count) ------------> would be 1
let parameter = parameterString.count > 0 ? parameterString[1] : "" // -----------> CRASH 
print(parameter)
```

Reply:
In that case ``` if route.contains(":") {``` redundant check needs to be added in the method  ..Please let me know if you want me to do that !?